index.js

```diff
@@ -38,14 +38,26 @@ function fastifyStatic (fastify, opts, next) {
     maxAge: opts.maxAge
   }
 
-  function pumpSendToReply (request, reply, pathname, rootPath) {
+  function pumpSendToReply (request, reply, pathname, rootPath, checkedExtensions) {
     var options = Object.assign({}, sendOptions)
 
     if (rootPath) {
       options.root = rootPath
     }
 
-    const stream = send(request.raw, pathname, options)
+    let encodingExtension = ''
+
+    if (opts.preCompressed) {
+      if (!checkedExtensions) {
+        checkedExtensions = new Set()
+      }
+      encodingExtension = checkEncodingHeaders(request.headers, checkedExtensions)
+    }
+
+    const pathnameWithExtension = pathname + `${encodingExtension ? `.${encodingExtension}` : ''}`
```

Comment:
Can you please avoid the ternary here and move the two values within the `preCompressed` if?